### PR TITLE
Support negative `axis` for `F.log_softmax`

### PR DESCRIPTION
**Pull request by toslunar**
This PR also allows 1-dim inputs.

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
LGTM!

The CI failures are not relevant to this change and should be handled as other issues:
- TestContrastive.test_double_backward_cpu
- TestTriplet.test_double_backward_cpu
